systemtest/src/test/java/io/strimzi/systemtest/upgrade/AbstractUpgradeST.java

```diff
@@ -147,26 +149,25 @@ protected void makeSnapshots() {
     }
 
     @SuppressWarnings("CyclomaticComplexity")
-    protected void changeKafkaAndLogFormatVersion(JsonObject procedures, JsonObject testParameters, ExtensionContext extensionContext) throws IOException {
+    protected void changeKafkaAndLogFormatVersion(Map<String, String> procedures, VersionModificationData testParameters, ExtensionContext extensionContext) throws IOException {
         // Get Kafka configurations
-        String operatorVersion = testParameters.getString("toVersion");
+        String operatorVersion = testParameters.getToVersion();
         String currentLogMessageFormat = cmdKubeClient().getResourceJsonPath(getResourceApiVersion(Kafka.RESOURCE_PLURAL, operatorVersion), clusterName, ".spec.kafka.config.log\\.message\\.format\\.version");
         String currentInterBrokerProtocol = cmdKubeClient().getResourceJsonPath(getResourceApiVersion(Kafka.RESOURCE_PLURAL, operatorVersion), clusterName, ".spec.kafka.config.inter\\.broker\\.protocol\\.version");
         // Get Kafka version
         String kafkaVersionFromCR = cmdKubeClient().getResourceJsonPath(getResourceApiVersion(Kafka.RESOURCE_PLURAL, operatorVersion), clusterName, ".spec.kafka.version");
         kafkaVersionFromCR = kafkaVersionFromCR.equals("") ? null : kafkaVersionFromCR;
-        String kafkaVersionFromProcedure = procedures.getString("kafkaVersion");
+        String kafkaVersionFromProcedure = procedures.get("kafkaVersion");
```

Comment:
If you will pass the procedures into the `VersionModificationData`, you will have getter here.